client/rest/src/main/java/org/elasticsearch/client/RestClient.java

```diff
@@ -77,7 +77,7 @@
  *
  * Requests can be traced by enabling trace logging for "tracer". The trace logger outputs requests and responses in curl format.
  */
-public final class RestClient implements Closeable {
+public class RestClient implements Closeable {
```

Comment:
> Do you think that is too much to ask?

I started down that path, but it was painful to mock out all of the raw client calls in a way that reasonably ensured that the parameters were supplied properly.

```java
client.execute(requestProducer, responseConsumer, new FutureCallback<HttpResponse>() {
```

Otherwise I was stuck capturing the producer _for every use_ and marching through it all, which is _far_ more cumbersome than "did we pass the right endpoint? the right params? headers? done."

The `RestClient` provides a really clean and succinct API that makes it incredibly easy to ensure validity without getting into the weeds.